sklearn/linear_model/_base.py

```diff
@@ -518,7 +527,16 @@ def fit(self, X, y, sample_weight=None):
             # Sample weight can be implemented via a simple rescaling.
             X, y = _rescale_data(X, y, sample_weight)
 
-        if sp.issparse(X):
+        if self.positive:
+            if y.ndim < 2:
+                self.coef_ = optimize.nnls(X, y)[0]
+            else:
+                # scipy.optimize.nnls cannot handle y with shape (M, K)
+                outs = Parallel(n_jobs=n_jobs_)(
+                        delayed(optimize.nnls)(X, y[:, j])
```

Comment:
Should we be giving the user control of maxiter?

Reply:
@jnothman, I have added `maxiter` as parameter: it turns out that `maxiter` landed on `optimize.nnls` starting from version 1.1.0 and scikit-learn still support scipy 0.19.1 in a number of platforms. Should I apply some special trick? Or I just revert this change? Thanks for your patience.

Reply:
I'm fine with not exposing it considering the version issues. If users want it, we can reconsider in the future

Reply:
@jnothman I've removed the `maxiter` control: please, feel free to comment if you don't agree with this choice. Thanks.